### PR TITLE
[Gekidou] [WIP] Data handlers

### PR DESCRIPTION
**Pull request by shazm**
#### Summary

Groups fetched during the autocomplete are saved locally.

The `@` mention in posting messages makes a call to `getGroupsForAutocomplete`; which usually returns a few groups. While this method still returns the `Promise<Group[]>`, it also passes this off to `getAssociationsForGroups`, where the Memberships (Users), Channels and Teams for each of the groups is fetched and saved locally.

To-do:
- [ ] Delete existing related data before saving newly fetched data for group associations.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-43983

#### Checklist

- [x] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 

#### Release Note

```release-note

```

### CODE REVIEW

**Comment by shazm** [comment]
@enahum / @avinashlng1080 / @larkox - This is still a WIP but I thought I'd get your thoughts now while it waits for the other PR's to merge.

File: app/actions/remote/groups.ts [hunk 1]
```diff
@@ -63,3 +70,66 @@ export const getChannelsForGroup = async (serverUrl: string, groupId: string) =>
...
+        models.push(...preparedGroupTeams);
+    }
+
+    const preparedGroupMemberships = await prepareGroupMemberships(operator, fetchedGroupMemberships);
+    if (preparedGroupMemberships.length) {
+        models.push(...preparedGroupMemberships);
+    }
+
+    try {
+        operator.batchRecords(models);
```
Comment (enahum) [inline app/actions/remote/groups.ts:126]: are we deleting groups that no longer exist or we are no longer part of ?

**Review by enahum — COMMENTED** [review]
high level review looks ok, need to do a more thorough review, but I would need the answer to my question first.

File: app/database/operator/server_data_operator/transformers/group.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    const record = value.record as GroupModel;
+    const isCreateAction = action === OperationType.CREATE;
+
+    // id of group comes from server response
+    const fieldsMapper = (group: GroupModel) => {
+        group._raw.id = isCreateAction ? (raw?.id ?? group.id) : record.id;
+        group.name = raw.name;
+        group.displayName = raw.display_name;
+
+        // group.source = raw.source;
```
Comment (larkox) [inline app/database/operator/server_data_operator/transformers/group.ts:40]: Is this needed any longer?

**Review by larkox — COMMENTED** [review]
In general looks nice. Just a few comments.

**Comment by shazm** [comment]
Closing this for smaller ones based on updated work; see https://mattermost.atlassian.net/browse/MM-43772

**Pull request closed without merging**